packages/iov-ethereum/src/ethereumconnection.ts

```diff
@@ -196,7 +233,62 @@ export class EthereumConnection implements BcpConnection {
   }
 
   public watchBlockHeaders(): Stream<BlockHeader> {
-    throw new Error("Not implemented");
+    // see https://github.com/ethereum/go-ethereum/wiki/RPC-PUB-SUB#newheads
+    this.socket!.send(
+      JSON.stringify({
+        id: "watchBlockHeaders",
+        jsonrpc: "2.0",
+        method: "eth_subscribe",
+        params: ["newHeads", { includeTransactions: true }],
+      }),
+    );
```

Comment:
Only send this when producer is started (`start: listener => {` block some lines below). Calling `watchBlockHeaders()` should not do heavy work before someone starts listening to the stream